### PR TITLE
add array length check

### PR DESCRIPTION
**Pull request by WanLinghao**
**What this PR does / why we need it**:
add array length check to avoid potential null point bug
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by WanLinghao** [comment]
@deads2k  PTAL

File: staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go [hunk 1]
```diff
@@ -182,7 +182,7 @@ func DefaultVerifyOptions() x509.VerifyOptions {
 
 // CommonNameUserConversion builds user info from a certificate chain using the subject's CommonName
 var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (user.Info, bool, error) {
-	if len(chain[0].Subject.CommonName) == 0 {
+	if len(chain) == 0 || len(chain[0].Subject.CommonName) == 0 {
```
Comment (nikhita) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go]: :+1:

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go]: error or nil?

**Pull request closed without merging**